mh/pat_item.py

```diff
@@ -585,6 +585,51 @@ class UserSearchInfo(PatData):
         (0x10, "info_mine_0x10"),
     )
 
+class LayerPath(object):
+    def __init__(self, server_id=None, gate_id=None, city_id=None):
+        # type: (int|None, int|None, int|None) -> None
+        self.server_id = server_id if server_id else 0
+        self.gate_id = gate_id if gate_id else 0
+        self.city_id = city_id if city_id else 0
+    
+    def get_depth(self):
+        # type: () -> int
+        if self.city_id > 0:
+            return City.LAYER_DEPTH
+        elif self.gate_id > 0:
+            return Gate.LAYER_DEPTH
+        elif self.server_id > 0:
+            return Server.LAYER_DEPTH
+        return -1
+
+    def pack(self):
+        # type: () -> bytes
+        depth = self.get_depth()
+        unk = 1
+        return struct.pack(">IIHHH", depth, self.server_id, unk, self.gate_id,
+                           self.city_id)
+    
+    @staticmethod
+    def unpack(data):
+        # type: (bytes) -> LayerPath
+        with Unpacker(data) as unpacker:
+            _ = unpacker.unpack_long()
+            server_id = unpacker.unpack_long()
```

Comment:
Unless I'm mistaken, I don't think this function will work. `pat_item.Unpacker` unpacks PAT items _(i.e. bytes with an `ItemType` identifier and its data)_, so `unpack_long` isn't the same as `struct.unpack(">I")`. You can use ` _, server_id = unpacker.struct("II")` to unpack like `struct.unpack`.